doc/tutorials/acrn-dm_QoS.rst

```diff
@@ -0,0 +1,121 @@
+.. _acrn-dm_qos:
+
+Enable QoS based on runC container
+##################################
+ACRN supports Device-Model QoS based on runC container to control
+the SOS resources (CPU, Storage, Memory, Network) by modifying the runC configuration file.
+
+What is QoS
+***********
+Traditionally, Quality of Service (QoS) is the description or measurement
+of the overall performance of a service, such as a `computer network
+<https://en.wikipedia.org/wiki/Computer_network>`_ or a `cloud computing
+<https://en.wikipedia.org/wiki/Cloud_computing>`_ service,
+particularly the performance is seen by the users on the network.
+
+What is runC container
+**********************
+Containers are an abstraction at the application layer that packages code
+and dependencies together. Multiple containers can run on the same machine
+and share the OS kernel with other containers, each running as
+isolated processes in user space. `runC
+<https://github.com/opencontainers/runc>`_, a lightweight universal container runtime,
+is a command-line tool for spawning and running containers according
+to the `Open Container Initiative (OCI)
+<https://www.opencontainers.org/>`_ specification.
+
+ACRN-DM QoS architecture
+************************
+In ACRN-DM QoS design, we run the ACRN-DM in a runC container environment.
+Every time we start a UOS, we run a runC container at first;
+then launch the ACRN-DM in the started container. The resources
+the ACRN-DM QoS can manage for Device-Model:
+
+- CPU utilization
+- Memory amount/limitation
+- I/O bandwidth
+- Network throughput
+
+.. figure:: images/acrn-dm_qos_architecture.png
+   :align: center
+
+   ACRN-DM QoS architecture
+
+ACRN-QoS CPU utilization example
+********************************
+In runC ``config.json`` we set the CPU resource as below for VM0 and VM1:
+
+.. code-block:: none
+
+    "cpu": {
+       "shares": 1024,
+       "quota": 1000000,
+       "period": 500000,
+       "realtimeRuntime": 950000,
+       "realtimePeriod": 1000000,
+       "cpus": "2-3",
+       "mems": "0-7"
+    },
+
+In this example the CPU shares value is 1024, so the VM0 or VM1 device model
+CPU utilization is ``1024 / (1024 + 1024 +1024) = 33%``, which means
+the maximal CPU resource for the VM0 or VM1 is 33% of the entire CPU resource.
+
+.. figure:: images/cpu_utilization_image.png
+   :align: center
+
+   CPU utilization image
+
+How to use ACRN-DM QoS
+**********************
+
+ #. Add these parameters to the ``runC.json`` file:
```

Comment:
This whole section seems to be indented by one extra space (and causes a problem in the generated output).